blog/tests.py

```diff
@@ -191,6 +191,28 @@ def test_no_past_upcoming_events(self):
         self.assertEqual(response.status_code, 200)
         self.assertQuerySetEqual(response.context["events"], [])
 
+    def test_no_unpublished_future_events(self):
+        """
+        Make sure there are no unpublished future events in the "upcoming events" sidebar
+        """
+        # We need a published entry on the index page so that it doesn't return a 404
+        Entry.objects.create(pub_date=self.yesterday, is_active=True, slug="a")
+        Event.objects.create(
+            date=self.tomorrow,
+            pub_date=self.yesterday,
+            is_active=False,
+            headline="inactive",
+        )
+        Event.objects.create(
+            date=self.tomorrow,
+            pub_date=self.tomorrow,
+            is_active=True,
+            headline="future publish date",
+        )
+        response = self.client.get(reverse("weblog:index"))
+        self.assertEqual(response.status_code, 200)
+        self.assertQuerySetEqual(response.context["events"], [])
```

Comment:
This new test already passes with the old version of the code, because it's not useing a logged-in staff user.
I suggest to test things with three types of users: anonymous (not logged in), logged in (not staff), and logged in (staff). Something like this should work (not tested):

```suggestion
        for user in [
            None,
            User.objects.create(is_staff=False),
            User.objects.create(is_staff=True),
        ]:
            if user:
                self.client.force_login(user)
        response = self.client.get(reverse("weblog:index"))
        with self.subTest(user=user):
            self.assertEqual(response.status_code, 200)
            self.assertQuerySetEqual(response.context["events"], [])
```

Reply:
Thank you for these tests! I didn't think to check the anonymous, staff, and non-staff users. I've added these cases along with a superuser for good measure!